### PR TITLE
use aggregated profiles for optimization

### PR DESCRIPTION
**Pull request by joAschauer**
@markushal I hope you can use this as a starting point. Tests of the optimization runs are still failing. I did not want to modify every single result in there, as I do not know whether it makes sense what is happening.

### CODE REVIEW

File: flh_opt/api_opt.py [hunk 1]
```diff
@@ -154,26 +153,27 @@ def add_storage(n: Network, input_data: dict, name: str, bus: str) -> None:
...
+        res_profiles, weights = get_profiles_and_weights(
             source_region_code=input_data["SOURCE_REGION_CODE"],
-            process_code=process_code,
             re_location=re_location,
             selection=range(0, 48),  # TODO: make this a function parameter?
-        )["specific_generation"]
+        )
 
     # define snapshots:
     n.snapshots = res_profiles.index
```
Comment (joAschauer) [inline flh_opt/api_opt.py:169]: `res_profiles` currently has a MultiIndex with names `["period_id", "TimeStep"]`. I am not sure if that is okay for the pypsa network.

Comment (markushal) [reply flh_opt/api_opt.py:169]: i fixed this.

File: flh_opt/api_opt.py [hunk 2]
```diff
@@ -154,26 +153,27 @@ def add_storage(n: Network, input_data: dict, name: str, bus: str) -> None:
...
 
     # define snapshot weightings:
-    n.snapshot_weightings["generators"] = 8760.0 / len(n.snapshots)
-    n.snapshot_weightings["objective"] = 8760.0 / len(n.snapshots)
+    if not not math.isclose(weights.sum(), 8760):
```
[the remark this answers is not in the document]
Comment (markushal) [reply flh_opt/api_opt.py]: I fixed this

**Review by markushal — APPROVED** [review]
Thanks! There were two issues which i fixed in my commits.

**Pull request merged**